packages/request-client.js/src/api/payment-network/erc20/proxy-contract.ts

```diff
@@ -0,0 +1,78 @@
+import { AdvancedLogicTypes, ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import * as Types from '../../../types';
+
+/**
+ * Handle payment networks with ERC20 proxy contract extension
+ */
+export default class PaymentNetworkERC20ProxyContract implements Types.IPaymentNetwork {
+  private extension: ExtensionTypes.PnReferenceBased.IReferenceBased;
+  /**
+   * @param extension The advanced logic payment network extensions
+   */
+  public constructor({ advancedLogic }: { advancedLogic: AdvancedLogicTypes.IAdvancedLogic }) {
+    this.extension = advancedLogic.extensions.proxyContractErc20;
+  }
+
+  /**
+   * Creates the extensions data for the creation of this extension.
+   * Will set a salt if none is already given
+   *
+   * @param paymentNetworkCreationParameters Parameters to create the extension
+   * @returns The extensionData object
+   */
+  public createExtensionsDataForCreation(
+    paymentNetworkCreationParameters: Types.IReferenceBasedCreationParameters,
+  ): ExtensionTypes.IAction {
+    // If no salt is given, generate one
+    const salt = paymentNetworkCreationParameters.salt || Utils.crypto.generate8randomBytes();
```

Comment:
`Utils.crypto.generate8randomBytes();`
Is it the right format? it should be string in 'hex', no? @romaric-juniet ?
if the function returns such a string the naming is bad (and it's probably my bad)

Reply:
![image](https://user-images.githubusercontent.com/11988817/70595314-09f6fe80-1c1e-11ea-9856-672407828080.png)

Reply:
it generates 8 random bytes and returns them as a string